planner/core/plan.go

```diff
@@ -446,6 +467,42 @@ func (*baseLogicalPlan) ExplainInfo() string {
 	return ""
 }
 
+func getEstimatedProbeCntFromProbeParents(probeParents []PhysicalPlan) float64 {
+	res := float64(1)
+	for _, pp := range probeParents {
+		switch pp.(type) {
+		case *PhysicalApply, *PhysicalIndexHashJoin, *PhysicalIndexMergeJoin, *PhysicalIndexJoin:
+			if join, ok := pp.(interface{ getInnerChildIdx() int }); ok {
+				outer := pp.Children()[1-join.getInnerChildIdx()]
+				res *= outer.statsInfo().RowCount
+			}
+		}
+	}
+	return res
+}
+
+func getActualProbeCntFromProbeParents(pps []PhysicalPlan, statsColl *execdetails.RuntimeStatsColl) int64 {
```

Comment:
Do we need this one? The actual row count should be correct now?

Reply:
Yes, this doesn't affect the display in the result of normal EXPLAIN.
The "true_card_cost" need this. It uses the real row count to calculate the cost. We need to convert the real row count to fit the assumption in our cost model.
The changes in `planner/core/plan_cost.go` uses this.